libs/ledger/src/protocols/notarisation_service.cpp

```diff
@@ -384,6 +384,21 @@ NotarisationService::AggregateSignature NotarisationService::GetAggregateNotaris
   return notarisation;
 }
 
+void NotarisationService::RemoveNotarisation(BlockNumber const &block_number,
+                                             BlockHash const &  block_hash)
+{
+  FETCH_LOCK(mutex_);
+
+  if (notarisations_being_built_.find(block_number) != notarisations_being_built_.end())
+  {
+    notarisations_being_built_[block_number].erase(block_hash);
```

Comment:
Would it be ok to have an empty map allowing just:

```
    notarisations_being_built_[block_number].erase(block_hash);
    notarisations_built_[block_number].erase(block_hash);
```

Looks like these aren't trimmed at any point - might be good to do similar to in the block coordinator (unless I misunderstand and they are trimmed via removenotarisation)

Reply:
They are trimmed but in PR #1901